streams/src/test/java/org/apache/kafka/streams/integration/RelaxedNullKeyRequirementJoinTest.java

```diff
@@ -71,11 +71,21 @@ void afterEach() {
     @Test
     void testRelaxedLeftStreamStreamJoin() {
         leftStream
-            .leftJoin(rightStream, JOINER, WINDOW)
+            .leftJoin(rightStream, JOINER, WINDOW_60MS_GRACE_10MS)
             .to(OUT);
         initTopology();
-        left.pipeInput(null, "leftValue", 1);
-        assertEquals(Collections.singletonList(new KeyValue<>(null, "leftValue|null")), out.readKeyValuesToList());
+        left.pipeInput(null, "leftValue1", 1);
+        left.pipeInput(null, "leftValue2", 90);
+        left.pipeInput(null, "lateArrival-Dropped", 19);
+        left.pipeInput(null, "lateArrivalWithinGrace", 20);
+        assertEquals(
+            Arrays.asList(
+                new KeyValue<>(null, "leftValue1|null"),
+                new KeyValue<>(null, "leftValue2|null"),
+                new KeyValue<>(null, "lateArrivalWithinGrace|null")
+            ),
+            out.readKeyValuesToList()
+        );
     }
 
     @Test
```

Comment:
Yes, I will check.

Similarily, I realized `KStreamKStreamSelfJoin` shoud probably also drop 'too late' records? I guess this would also be  a separate Jira ticket & PR?

Reply:
Yes it should, and yes, let's keep it separate -- if you want to extend the jira for self-joins or create a new one does not matter too much. We can also so more than one PR per Jira -- but this PR is already complex enough, so let's not add the self-join stuff, but only tackle it after we merged this one.